camviewer_ui_impl.py

```diff
@@ -2712,4 +2768,25 @@ def getConfig(self):
             self.setOrientation(int(orientation))
         except:
             pass
+        # Process projection settings, if any.
+        try:
+            self.ui.checkBoxProjRoi.setChecked(self.cfg.projroi == "1")
+            l = [ll == "1" for ll in self.cfg.projlineout]
+            self.ui.checkBoxM1Lineout.setChecked(l[0])
+            self.ui.checkBoxM2Lineout.setChecked(l[1])
+            self.ui.checkBoxM3Lineout.setChecked(l[2])
+            self.ui.checkBoxM4Lineout.setChecked(l[3])
+            self.ui.checkBoxFits.setChecked(self.cfg.projfit == "1")
+            l = [ll == "1" for ll in self.cfg.projfittype]
+            if l[0]:
+                self.ui.radioGaussian.setChecked(True)
+            if l[1]:
+                self.ui.radioSG4.setChecked(True)
+            if l[2]:
+                self.ui.radioSG6.setChecked(True)
+            self.calib = float(self.cfg.projcalib)
+            self.ui.lineEditCalib.setText(str(self.calib))
+        except:
```

Comment:
`try:` with a really big block and a bare `except: pass` can make debugging weird issues pretty frustrating...

`except:` is generally recommended to be `except Exception:` instead, because the former will eat up things like `KeyboardInterrupt` but the latter will not. I'd imagine this try block isn't intending to catch that - see [exception hierarchy](https://docs.python.org/3/library/exceptions.html#exception-hierarchy))

Consider maybe at least dumping out a traceback with a `logger`:

```python
try:
    do_the_thing
except Exception:
    logger.exception("Failed to do_the_thing")
```

That assumes a logger exists at module-level like `import logging; logger = logging.getLogger(__name__)` or some other way. Alternatively, an `import traceback` and in the except block `traceback.print_exc()` could be better than nothing.

Reply:
At this point, there is no logger.

And this block is in my state restore code anyway... so if it fails, I don't really care.  That's why I didn't really make any effort to try to figure out exactly why it failed and debug it... just move on and start displaying the image.

Reply:
I'll change all of the bare excepts to except Exception though.